phoenix-core/src/main/java/org/apache/phoenix/schema/types/PUnsignedLong.java

```diff
@@ -95,6 +95,11 @@ public Object toObject(byte[] b, int o, int l, PDataType actualType, SortOrder s
   }
 
   @Override
+    public boolean isCastableTo(PDataType targetType) {
```

Comment:
The problem with that is the smaller. Numbers in the jira like 22847 wind up being integer and not long so without doing both you have an arbitrarily small number where things just start working so cast(12949 as timestamp) will not work but cast(284484849 as timestamp) will. 

> On Mar 18, 2015, at 7:00 PM, James Taylor <notifications@github.com> wrote:
> 
> In phoenix-core/src/main/java/org/apache/phoenix/schema/types/PUnsignedLong.java:
> 
> > @@ -95,6 +95,11 @@ public Object toObject(byte[] b, int o, int l, PDataType actualType, SortOrder s
> >    }
> >  
> >    @Override
> > +    public boolean isCastableTo(PDataType targetType) {
> Was thinking to only make this change so that an UNSIGNED_LONG is castable to a TIMESTAMP. Allowing an INTEGER to be cast to a TIMESTAMP seems weird to me. How about we just start with this change for now and you adjust your tests accordingly?
> 
> —
> Reply to this email directly or view it on GitHub.
>